backend/src/hatchling/metadata/core.py

```diff
@@ -193,8 +203,9 @@ def core(self) -> CoreMetadata:
 
                 if metadata.dynamic:
                     for metadata_hook in metadata_hooks.values():
-                        metadata_hook.update(self.core_raw_metadata)
-                        metadata.add_known_classifiers(metadata_hook.get_known_classifiers())
+                        with set_version(metadata_hook, self.version):
```

Comment:
Here the context manager  - and we have version property in the hook which we set (and can query in the hook).